### PR TITLE
feat: Support multiple vector search

### PR DESCRIPTION
**Pull request by xige-16**
issue #25639 

Signed-off-by: xige-16 <xi.ge@zilliz.com>

### CODE REVIEW

**Comment by xige-16** [comment]
rerun ut

**Comment by xige-16** [comment]
rerun ut

File: internal/core/src/segcore/SegmentSealedImpl.cpp [hunk 1]
```diff
@@ -912,6 +912,17 @@ SegmentSealedImpl::check_search(const query::Plan* plan) const {
     AssertInfo(plan->extra_info_opt_.has_value(),
                "Extra info of search plan doesn't have value");
 
+    auto& metric_str = plan->plan_node_->search_info_.metric_type_;
```
Comment (czs007) [inline internal/core/src/segcore/SegmentSealedImpl.cpp:915]: Is it possible to abstract the above code into a function or place it in a base class?

File: internal/querynodev2/services.go [hunk 2]
```diff
@@ -253,8 +246,7 @@ func (node *QueryNode) WatchDmChannels(ctx context.Context, req *querypb.WatchDm
 
 	node.manager.Collection.PutOrRef(req.GetCollectionID(), req.GetSchema(),
 		node.composeIndexMeta(req.GetIndexInfoList(), req.Schema), req.GetLoadMeta())
-	collection := node.manager.Collection.Get(req.GetCollectionID())
-	collection.SetMetricType(req.GetLoadMeta().GetMetricType())
```
Comment (czs007) [inline internal/querynodev2/services.go:257]: Why remove this logic?
Previously, this logic was implemented to support the ability to search without specifying the metric_type.
The metric_type is already present in the loaded index, so users do not need to specify it

**Pull request merged**